### PR TITLE
Query: Relational: Adds nav prop to nav prop/null comparison translat…

### PR DESCRIPTION
**Pull request by anpete**
…ions.

### CODE REVIEW

**Comment by anpete** [comment]
@maumar

File: src/EntityFramework.Core/Query/ExpressionVisitors/NavigationRewritingExpressionVisitor.cs [hunk 1]
```diff
@@ -99,6 +113,70 @@ protected override Expression VisitConstant(ConstantExpression constantExpressio
...
+                    if (constantExpression != null
+                        && constantExpression.Value == null)
+                    {
+                        newRight = rightNavigationJoin.JoinClause.OuterKeySelector;
+
+                        NavigationJoin.RemoveNavigationJoin(_navigationJoins, rightNavigationJoin);
+                    }
+                    else
+                    {
+                        newRight = rightNavigationJoin.JoinClause.InnerKeySelector;
```
Comment (maumar) [inline src/EntityFramework.Core/Query/ExpressionVisitors/NavigationRewritingExpressionVisitor.cs]: are we guaranteed not to have convert on the right side of the binary op?

**Comment by maumar** [comment]
:shipit:

File: src/EntityFramework.Core/Query/ExpressionVisitors/NavigationRewritingExpressionVisitor.cs (hunk 1 above)
Comment (anpete) [inline src/EntityFramework.Core/Query/ExpressionVisitors/NavigationRewritingExpressionVisitor.cs]: Removed the RemoveConvert from above as it was no longer needed.

**Pull request merged**